cryptol-saw-core/src/Verifier/SAW/Cryptol.hs

```diff
@@ -1667,6 +1708,8 @@ tNoUser initialTy =
         if null params then C.TRec (C.ntFields fs)
                        else panic "tNoUser" ["Nominal type with parameters"]
                         -- XXX: We should instantiate, see #2019
+                        -- XXX: What about Enums?
+                        --   Which also have type params, #2019 again.
```

Comment:
No, I don't think we need a case for enums here. Newtypes are a bit special in that they are compiled away to SAWCore records, without any trace of the original newtype constructor. We can't do something like that for enums, so no need for special treatment here.